### ISSUE TITLE
ENH: sign convention for hessian_factor

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#4714]
I'm getting confused about the sign of hessian_factor. 

I was adding hessian factor to discrete so that `hessian = sum_i(hessian_factor_i * outer(exog_i))`, in an upcoming PR

However, GLM hessian_factor has the opposite sign
hessian =
```
factor = self.hessian_factor(params, scale=scale, observed=observed)
hess = -np.dot(self.exog.T * factor, self.exog)
``` 

I don't remember why GLM has that sign. The only reason that I can think of is so that hessian_factor is positive. My guess this came from SPSS.

The recently merged score_test assumes that the GLM definition prevails and raises with a opposite sign Poisson.

**Comment by thequackdaddy** [comment]
This may be related, but we have differing signs for hessian and score when you are using `newton` vs other optimizers

https://github.com/statsmodels/statsmodels/blob/47315ce8aaa7b0034c4a31c43c685cf1a44791fe/statsmodels/base/model.py#L442

**Comment by josef-pkt** [comment]
> This may be related, but we have differing signs for hessian and score when you are using newton vs other optimizers

I guess that might be more related to the definition of the optimizer, i.e. the sign convention used in our home-made "newton" in contrast to the scipy minimizers.
Our "newton" iteration might have been written for a maximization problem.

**Comment by jbrockmendel** [comment]
Is resolving this inconsistency an option?  It's affect on the code-base is expanding (#5711)

**Comment by josef-pkt** [comment]
It cannot be changed in a backwards compatible way, and I'm not sure what the consequences are.
So reverting the sign is not a quickfix. My guess is that we have the same problem in OLS and similar.

In GAM I ran also into problems with a factor of 2, e.g. OLS versus MLE objective function, or min sum of squares versus min 0.5 * sum of squares. OLS/WLS moment conditions do not include the 2 for the former.

The main underlying problem is that estimators based on optimization problems differ from estimators that use first order conditions (moment conditions or estimating equations). The latter are not required to have a fixed sign. e.g. #5674

bottom line: We need to figure out what the generic patterns should be before fixing existing algorithms.

For the definition of llf and loglike, we decided to use only the full version and not to drop any terms that are irrelevant in the optimization problem.
By the same argument we could fix the sign and terms in the score and hessian to be consistent with the optimization problem when there are overlapping cases between optimization and estimating/moment equations.

**Comment by josef-pkt** [comment]
For moment tests and score_test all we need is consistent sign between score and hessian.
That is sign convention for moment estimators, estimating equations, GMM should be irrelevant for it, as long as those signs are consistent.

OLS/WLS/GLS has hessian_factor but no score_factor
hessian_factor was added as part of `distributed` PR https://github.com/statsmodels/statsmodels/commit/44e1c728134317d5464b0604d84f432e3a765b01#diff-8cb6b075cf3ba145f963655324052ee2647c3554d9bb80e920ac520c8efe17bc
